Index: external/mit/lua/usr.bin/lua/Makefile
===================================================================
RCS file: /cvsroot/src/external/mit/lua/usr.bin/lua/Makefile,v
retrieving revision 1.5
diff -u -r1.5 Makefile
--- external/mit/lua/usr.bin/lua/Makefile	19 Jan 2011 11:55:50 -0000	1.5
+++ external/mit/lua/usr.bin/lua/Makefile	9 Sep 2015 18:10:19 -0000
@@ -4,10 +4,10 @@
 SRCS=		lua.c
 BINDIR=		/usr/bin
 
-CPPFLAGS+=	-DLUA_USE_READLINE
+CPPFLAGS+=	-DLUA_USE_POSIX
 
-LDADD+=		-llua -lm -ledit -lterminfo
-DPADD+=		${LIBLUA} ${LIBM} ${LIBEDIT} ${LIBTERMINFO}
+LDADD+=		-llua -lm 
+DPADD+=		${LIBLUA} ${LIBM}
 
 .include <bsd.own.mk>
 
